### PR TITLE
Release 1.21.2

### PR DESCRIPTION
**Pull request by kingdonb**
I have rebased and merged in #3325 and #3282 for inclusion into this release, which are both not already included in the `master` branch. The other PRs listed in the changelog have already been merged.

Both #3325 and #3282 should be inspected by a maintainer before merging this.

This is my first Flux daemon release. I reviewed [the document](https://github.com/fluxcd/flux/blob/master/internal/docs/releasing.md) and I think the procedure after this PR passes all CI checks, is for a maintainer to review, approve, and merge it. Then create the release in GitHub (which triggers CI to build images and push binaries with the 1.21.2 version)

Then I, or someone else, can create another PR for merge from release.1.21.x back to master (housekeeping), and a maintainer can push the tag for `chart-1.6.2`, finalizing the release. I have already been shown the checkpoint database and can complete the update for that as well.

Closes #3325 
Closes #3282

### CODE REVIEW

**Comment by yebyen** [comment]
#3296 could also use a review, as I could not understand if this was a breaking change or not. It is not included in this release though it looked like a candidate for next tag.

**Comment by kingdonb** [comment]
This should be replaced with 9e5a366 11 minutes ago from https://github.com/kingdonb/flux/tree/release/1.21.2 – it has been rebased and the `CHANGELOG.md` was updated.

That new commit incorporates #3423 that @dholbach added this morning, and was merged into master already.

(I do not have write access so I cannot update the branch `release/1.21.2` at fluxcd/flux myself)

**Comment by kingdonb** [comment]
The failure is a flaky test added in #3282. I'm running it down to see if I can understand why it is flaky.

**Comment by kingdonb** [comment]
This should be replaced with 87451be
from https://github.com/kingdonb/flux/tree/release/1.21.2 – it has been rebased and the CHANGELOG.md was updated.

I removed #3282 in that commit, since it has a flaky test at least, merging it as-is will result in incidents.

I do not have write access so I cannot update the branch release/1.21.2 at fluxcd/flux myself, however it was voted in the FluxCD meeting this morning to grant write access to myself to forward the purpose of simplifying this process 👍 🎉 

I will keep working with the maintainers for their approval on each release, but this removes the major obstacle for me which was being able to verify by myself that CI is passing and everything is good with the release on FluxCD's github org **before** I go out and tap a maintainer on the shoulder asking for their approval. Thanks @hiddeco, @squaremo, and @stefanprodan 👍 

Once the write access is granted, I can replace this PR and we should see the flaky test has been resolved.

**Comment by kingdonb** [comment]
The CI has passed, this release has my seal of approval 👍

I moved #3282 to the next milestone since it looks like it could be important, but I don't want to merge in a flaky test.

File: CHANGELOG.md
Comment (hiddeco) [inline CHANGELOG.md:11]: This will need an update before the release (NB: release dates are in UTC).

File: CHANGELOG.md
Comment (hiddeco) [inline CHANGELOG.md:25]: This is not related to the release of Flux and can be omitted.

File: CHANGELOG.md
Comment (kingdonb) [reply CHANGELOG.md:11]: Thanks. cef604e5 resolves this

File: CHANGELOG.md
Comment (kingdonb) [reply CHANGELOG.md:25]: Thanks. Dropped this from cef604e5 which I've just force-pushed over, it has this.

**Comment by kingdonb** [comment]
Just waiting for CI to pass again before merge. `1.21.2` should be released within the hour, and the chart published shortly thereafter.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**
